src/lager_exchange_backend.erl

```diff
@@ -187,10 +187,11 @@ maybe_init_exchange(State) ->
 %% @private
 init_exchange(true) ->
     {ok, DefaultVHost} = application:get_env(rabbit, default_vhost),
-    VHost = rabbit_misc:r(DefaultVHost, exchange, ?LOG_EXCH_NAME),
+    Exchange = rabbit_misc:r(DefaultVHost, exchange, ?LOG_EXCH_NAME),
     try
-        #exchange{} = rabbit_exchange:declare(VHost, topic, true, false, true, [], ?INTERNAL_USER),
-        {ok, #resource{virtual_host=DefaultVHost, kind=exchange, name=?LOG_EXCH_NAME}}
+        %% durable
+        #exchange{} = rabbit_exchange:declare(Exchange, topic, true, false, false, [], ?INTERNAL_USER),
```

Comment:
I tried a `rabbit_exchange:update/2` here to simplify upgrades but it requires a Mnesia transaction and at logger initialisation stage this would not be safe. So users who already had the exchange must delete it manually during upgrades. Failures are handled in this function so node startup is not at risk.